src/steps/05-PerformanceRequirements/DOW/ComputeForm.spec.ts

```diff
@@ -0,0 +1,700 @@
+            })
+          });
+        });
+        describe("tests validation on 'periodsCheckboxes(other option)' & " +
+                 "triggers error if ...", () => {
+          it("no data, then expect data.errorMessages.toHaveLength(1)", async()=>{
+            computeData.entireDuration = "NO";
+            computeData.periodsNeeded = [];
+   
+            await wrapper.setData({
+              _computeData: computeData,
+              availablePeriodCheckboxItems: availablePeriodCheckboxItems
+            });
+  
+            const spy = await jest.spyOn(wrapper.vm, "setErrorMessages");
+            await wrapper.vm.setErrorMessages();
+            expect(spy).toHaveBeenCalledTimes(1);
+            
+            //updated wrapper.setData causes periodCheckboxList to display
+            const periodCheckboxes = await wrapper.findComponent(
+              {ref:"periodsCheckboxes"}
+            );
+            await periodCheckboxes.setData({
+              prevSelected: []
+            })
+            expect(periodCheckboxes.exists()).toBe(true);
+            expect(periodCheckboxes.vm.$data.errorMessages).toHaveLength(1);
+          });
+        });
+        describe("tests validation on 'regionsCheckboxes(other option)' & " +
+                 "triggers error if ...", () => {
+          it("no data, then region checkbox html will contain expected error", async()=>{
+            computeData.deployedRegions = ["OtherRegion"];
+            computeData.deployedRegionsOther = "";
+   
+            await wrapper.setData({
+              _computeData: computeData,
+              otherRegionValue: "OtherRegion",
+              regionCheckboxOptions: regionCheckboxOptions
+            });
+          
+            const spy = await jest.spyOn(wrapper.vm, "setErrorMessages");
+            await wrapper.vm.setErrorMessages();
+            expect(spy).toHaveBeenCalledTimes(1);
+            
+            const regionCheckboxes = await wrapper.findComponent(
+              {ref:"regionsCheckbox"}
+            );
+            expect(regionCheckboxes.exists()).toBe(true);
+            expect(regionCheckboxes.html().trim()).toContain(
+              "<div class=\"field-error ml-2\">Please enter your other region(s).</div>"
+            );
+            
+          });
+        });
+      });
+
+      describe("various form controls...", ()=>{
+        it("testing OperatingSystemAndLicensing textbox 'required' rule, then setting " +
+           "data.errorMessages on textbox.blur()", async () => {
+          computeData.operatingSystemAndLicensing = "dummy text";
+          await wrapper.setData({
+            _computeData: computeData,
+          });
+
+          const textbox = await wrapper.findComponent({ref: "operatingSystemAndLicensing" });
+          expect(textbox.exists()).toBe(true);
+          expect(textbox.vm.$data.errorMessages.length).toBe(0)
+
+          computeData.operatingSystemAndLicensing = "";
```

Comment:
This should probably be in its own method since it's an entirely new set of "given, when, expect" clauses.

Reply:
The reason it is here is b/c of the data reactivity (compare lines 616 and 625). 

Good data does not generate an error message (lines 616-623) and invalid data does (lines 625-630)

Reply:
made 2 functions out of this one function.  The functions are differentiated as one handles valid data and the other one handles invalid data.